contracts/core/Allo.sol

```diff
@@ -1,128 +1,70 @@
- *      /::\~\:\  \   /:/  /    /:/  /    /:/  \:\  \
- *     /:/\:\ \:\__\ /:/__/    /:/__/    /:/__/ \:\__\
- *     \/__\:\/:/  / \:\  \    \:\  \    \:\  \ /:/  /
- *          \::/  /   \:\  \    \:\  \    \:\  /:/  /
- *          /:/  /     \:\  \    \:\  \    \:\/:/  /
- *         /:/  /       \:\__\    \:\__\    \::/  /
- *         \/__/         \/__/     \/__/     \/__/
+ *          ___            ___        ___        ___
+ *         /\  \          /\__\      /\__\      /\  \
+ *        /::\  \        /:/  /     /:/  /     /::\  \
+ *       /:/\:\  \      /:/  /     /:/  /     /:/\:\  \
+ *      /::\~\:\  \    /:/  /     /:/  /     /:/  \:\  \
+ *     /:/\:\ \:\__\  /:/__/     /:/__/     /:/__/ \:\__\
+ *     \/__\:\/:/  /  \:\  \     \:\  \     \:\  \ /:/  /
+ *          \::/  /    \:\  \     \:\  \     \:\  /:/  /
+ *          /:/  /      \:\  \     \:\  \     \:\/:/  /
+ *         /:/  /        \:\__\     \:\__\     \::/  /
+ *         \/__/          \/__/      \/__/      \/__/
  */
 
 /// @title Allo
 /// @notice The Allo contract
 /// @author allo-team
-contract Allo is Transfer, Initializable, Ownable, AccessControl {
-    using SafeERC20Upgradeable for IERC20Upgradeable;
-
-    /// @notice Custom errors
-    error UNAUTHORIZED();
-    error NOT_ENOUGH_FUNDS();
-    error NOT_APPROVED_STRATEGY();
-    error IS_APPROVED_STRATEGY();
-    error MISMATCH();
-    error ZERO_ADDRESS();
-    error INVALID_FEE();
-
-    /// @notice Struct to hold details of an Pool
-    struct Pool {
-        bytes32 identityId;
-        IStrategy strategy;
-        address token;
-        uint256 amount;
-        Metadata metadata;
-        bytes32 managerRole;
-        bytes32 adminRole;
-    }
-
+contract Allo is IAllo, Transfer, Initializable, Ownable, AccessControl {
     /// @notice Fee denominator
     uint256 public constant FEE_DENOMINATOR = 1e18;
 
     /// ==========================
     /// === Storage Variables ====
     /// ==========================
 
+    address constant NATIVE = address(0);
     /// @notice Fee percentage
     /// @dev 1e18 = 100%, 1e17 = 10%, 1e16 = 1%, 1e15 = 0.1%
-    uint256 public feePercentage;
+    uint256 private feePercentage;
 
     /// @notice Base fee
-    uint256 public baseFee;
+    uint256 internal baseFee;
 
     /// @notice Incremental index
     uint256 private _poolIndex;
 
     /// @notice Allo treasury
-    address payable public treasury;
+    address payable private treasury;
 
     /// @notice Registry of pool creators
-    Registry public registry;
+    IRegistry private registry;
 
     /// @notice msg.sender -> nonce for cloning strategies
     mapping(address => uint256) private _nonces;
 
     /// @notice Pool.id -> Pool
-    mapping(uint256 => Pool) public pools;
+    mapping(uint256 => Pool) private pools;
 
     /// @notice Strategy -> bool
-    mapping(address => bool) public approvedStrategies;
+    mapping(address => bool) private approvedStrategies;
```

Comment:
Are we still using this? Remember we decided to remove the idea of approved strategies - can we remove this?

Reply:
for now it isfor now it is still an open question
and since it was already implemented, we haven't removed it.

i'm still against removing it :D - but once the final decision is made, we can touch the code again